### PR TITLE
Have TestServer merge the user-agent header 

### PR DESCRIPTION
**Pull request by Tratcher**
#18198
User-Agent defies the normal HTTP rules as it's explicitly space delimited (not commas). HttpClient normally deals with this by overriding the separator (only user-agent does this).
https://github.com/dotnet/runtime/blob/aa13fb88d7640f1ccee952976ea45345c97cd7a3/src/libraries/System.Net.Http/src/System/Net/Http/Headers/ProductInfoHeaderParser.cs#L13-L14
https://github.com/dotnet/runtime/blob/4f9ae42d861fcb4be2fcd5d3d55d5f227d30e723/src/libraries/System.Net.Http/src/System/Net/Http/Headers/HttpHeaders.cs#L319-L323
Unfortionately those APIs are internal, we have to add our own special case.

### CODE REVIEW

**Review by halter73 — APPROVED** [review]

**Comment by anurse** [comment]
@msftbot wait until 10AM PST to merge this

**Comment by anurse** [comment]
Meh, was just trying something :)

**Pull request merged**
